best-practices/sdk-principles.md

```diff
@@ -0,0 +1,226 @@
+and autonomy of teams working on new client software (SDKs), whilst also ensuring where possible/practicable, we have consistency in, amongst other things, the API, our testing strategy,
+versioning approach, developer tools, observability, debugging and integration with the core SDKs.
+
+## Product strategy context
+
+Before we dive into the client software best practices, it’s important we are all aligned on our product strategy.
+In our 22nd NTK of 2023, we present an [SLT Strategy update](https://ably.atlassian.net/wiki/spaces/PEO/pages/2461434058/2023+Need+to+knows#%2322-2023-Need-to-know-(29th-of-June-2023)---SLT-Strategy-Update),
+which included a section on how our product strategy is evolving. If you are unclear on the product strategy, please listen to the recording for slides 12 → 31
+(11m 50s → 37m-ish).  The TLDR version as of June 2023 is summarised in slide 30 below:
+
+![Product Strategy](/images/product-strategy.png)
+
+1. **Ably’s Core product is Pub/Sub Channels**. This is the product that we launched back in 2016, and has continued to be the driver for developers to deliver any type of realtime functionality they require in their applications. The core product and the associated SDKs are considered to be a set of lower level API primitives that can power anything from Chat, to Multiplayer, to Broadcast, but typically a developer needs to handle the application space logic (like how their Chat works), whilst we handle the the lower level functionality reliably and at scale, such as pub/sub messaging, history, presence, authentication, etc.
+
+2. Ably’s **newer products** aim to solve a **use case or more specific technical problem**. Take for example Spaces (our multiplayer offering). Whilst it is is built on to of Pub/Sub channels, it exposes a set of APIs to solve specific requirements for developers who want to add multiplayer collaboration to their apps. As such, we offer a Locks API, a Locations API, and a Cursor Tracking API. The complexity of these new SDKs is not necessarily high, but that’s less important quite frankly.  What is important is that a developer can, for example, be abstracted away from how live cursors works and can use an API that is specifically designed to help them solve that specific problem. Our goal with our higher level API product offerings is to take away more of the complexity from developers, and offer purpose built APIs that make it easy for developers to work on their application logic, as opposed to common use case logic relating to realtime experiences, such as delivering chat, or keeping the application state in sync with the server, or showing what other participants are doing inside a collaborative application. 
+
+Please see slide 29 from the SLT strategy update for considerations we need to bear in mind:
+
+![Product Consideration](/images/product-consideration.png)
+
```

Comment:
is [this part](https://ably.atlassian.net/wiki/spaces/ENG/pages/2692874253/Client+software+SDK+principles+and+best+practice+RFC#:~:text=On%20whether%20each%20product%20needs%20an%20SDK%20or%20not%3A) from the original document was intentionally omitted:
```
1. On whether each product needs an SDK or not:

a. Multiplayer & Spaces, as of Aug 2023, are providing new SDKs

b. Ably Asset Tracking already has its own set of SDKs

c. Conversations (Chat) functionality and Notifications are currently planned to be part of the core pub/sub channel SDKs, but that is subject to change as this evolves.
```
?

Reply:
No, by accident, fixed